### PR TITLE
ci: added priority rules to mergify configuration

### PR DESCRIPTION
**Pull request by frazarshad**
## Description

Mergify has recently added priority rules feature to mergify configuration

This allows us to define priority rules of each item in our merge queue, independent of the the priority of the queue

What this means for us is that we can remove all the separate high priority queues and simplify our configuration

Also created a [PR in Agoric/mergify-experiments](https://github.com/Agoric/mergify-experiements/pull/25) that the updates made here

### Testing

This configuration has been tested out on [frazarshad/mergify-experiments ](https://github.com/frazarshad/mergify-experiements)
image

Similarly I also tested that the queues don't have any priority base on their order in the configuration and don't preempt each other
image

### CODE REVIEW

**Review by michaelfig — APPROVED** [review]
LGTM!

I'm prepared to maintain these changes to `.mergify.yml`, but feel free to wait for a review by @mhofman if you wish.

**Review by mhofman — APPROVED** [review]
Just to make sure I understand the upstream change. There is now a reified priority concept in mergify, and queue rules no longer have an implicit priority derived from order, so we can drop all the `disallow_checks_interruption_from_queues`. We have verified that adding a  "merge" label on a PR will not interrupt a "rebase" PR that is already in the queue?

**Comment by frazarshad** [comment]
> Just to make sure I understand the upstream change. There is now a reified priority concept in mergify, and queue rules no longer have an implicit priority derived from order, so we can drop all the `disallow_checks_interruption_from_queues`. We have verified that adding a "merge" label on a PR will not interrupt a "rebase" PR that is already in the queue?

@mhofman its definitely a good idea to check that. I've verified that a queue does not have any priority based on its ordering in the configuration.
I've added a screenshot in the PR ddescription

**Pull request merged**
